roles/confluent.zookeeper/tasks/main.yml

```diff
@@ -97,6 +112,14 @@
     owner: "{{zookeeper_user}}"
     group: "{{zookeeper_group}}"
 
+- name: Create Zookeeper Config directory
```

Comment:
is this task necessary? My understanding is `zookeeper.config_file | dirname` resolves to:
 `{{ config_base_path }}/etc/kafka/zookeeper.properties | dirname` which resolves to:
`opt/confluent/whatever/etc/kafka/'

And shouldn't that dir be created when the tar is unarchived?

Reply:
Not entirely correct. The configuration is kept separate from the expanded tarball. For example, a tarball may be expanded to `/opt/confluent`, which results in the path `/opt/confluent/confluent-5.5.1`. However, the configuration would be kept in `/opt/confluent/etc`. I did this so different versions could be expanded, but potentially use the same configuration files. Also, it allows another expansion to happen if for some reason the original gets corrupted. Maybe I'm over-thinking these points, but that's what was going through my head.

Reply:
Thoughts on this approach?

Reply:
I put some thought haha and couldnt really come up with a good reason why its wrong. In the package installation option we do overwrite the server.properties... and by this logic maybe we should save it into a separate file so that when updating packages the properties files doesnt get overwrite... but thats potential tech debt haha! I'm cool with this approach

Reply:
Heres something i just thought of, curious your take.

The normal path for kafka's server.properties is `/etc/kafka/server.properties` why not put it there instead of `/opt/confluent/etc/kafka/server.properties` 

There is already the tasks that creates the directory to put the properties file into, so you would only have to change the variable definition

Reply:
The original intent of putting it in `/opt` was to have it in a place you didn't need to be root to access. Also, if someone accidentally runs a yum/apt install then those files will be overwritten. We can default to `/etc/whatever`, but I would really want to keep this as a variable to override, i.e. the existing `config_base_path` in `roles/confluent.variables/vars/main.yml`